docs/standard/library-guidance/dependencies.md

```diff
@@ -0,0 +1,95 @@
+
+## Diamond dependencies
+
+It's a common situation for a .NET project to have multiple versions of a package in its dependency tree. For example, an app depends on two NuGet packages, each of which depends on different versions of the same package. A diamond dependency now exists in the app's dependency graph.
+
+![Diamond dependency](./media/diamond-dependency.png "Diamond dependency")
+
+At build time NuGet analyzes all the packages that a project depends on, including the dependencies of dependencies, and when multiple versions of a package are detected, rules are evaluated to pick one. Unifying packages is necessary because running side-by-side versions of an assembly in the same application is problematic in .NET.
+
+Most diamond dependencies are easily resolved, however they can create issues in certain circumstances:
+
+1. **Conflicting NuGet package references** prevent a version from being resolved during package restore.
+2. **Breaking changes between the versions** cause bugs and exceptions at runtime.
+3. **The package assembly is strong named**, the assembly version changed, and the app is running on the .NET Framework. Assembly binding redirects are required.
+
+It's not possible to know what packages will be used alongside your own. A good way to reduce the likelihood of a diamond dependency breaking your library is to minimize the number of packages you depend on.
+
+**✔️ DO** review your .NET library for unnecessary dependencies.
+
+## NuGet dependency version ranges
+
+A package reference specifies the range of valid packages it allows. Typically, the package reference version in the `csproj` file is the minimum version and there's no maximum.
+
+```xml
+<!-- Accepts any version 1.0 and above. -->
+<PackageReference Include="ExamplePackage" Version="1.0" />
+```
+
+The rules that NuGet uses when resolving dependencies are [complex](https://docs.microsoft.com/en-us/nuget/consume-packages/dependency-resolution), but NuGet always looks for the lowest applicable version. NuGet prefers the lowest application version over using the highest available because the lowest will have the least compatibility issues.
+
+Because of NuGet's lowest application version rule, it isn't necessary to place an upper version or exact range on package references to avoid getting the latest version. NuGet already tries to find the lowest, most compatible version for you.
+
+```xml
+<!-- Accepts 1.0 up to 1.x, but not 2.0 and higher. -->
+<PackageReference Include="ExamplePackage" Version="[1.0,2.0)" />
+
+<!-- Accepts exactly 1.0. -->
+<PackageReference Include="ExamplePackage" Version="[1.0]" />
+```
+
+Upper version limits will cause NuGet to fail if there's a conflict. For example, one library accepts exactly 1.0 while another library requires 2.0 or above. While breaking changes may have been introduced in version 2.0, a strict or upper limit version dependency guarantees an error.
```

Comment:
As an aside, this is such a weird contrast to me between the behavior on L41 ("We will intentionally choose old software in order to save you from potential problems"), and the recommendation here ("Don't be strict about SemVer, just roll the dice, it might work!") - of course, if you're following the rest of the good advice here about avoiding breaking people at all, it'll be fine?

Reply:
L41 is context for why you shouldn't place an upper range on the package number. By default you're getting the most compatible version. You don't need to force it, and if you do you're likely going to create conflicts.